api/src/main/java/org/openmrs/module/reporting/definition/library/AllDefinitionLibraries.java

```diff
@@ -31,9 +32,42 @@
 @Repository
 public class AllDefinitionLibraries {
 
+	/**
+     * Modules can define beans that implement DefinitionLibrary
+     */
+    @Autowired(required = false)
+    private List<DefinitionLibrary<?>> libraryBeans;
+
+	/**
+     * Modules can define beans that implement DefinitionLibraryFactory
+     */
     @Autowired(required = false)
+    private List<DefinitionLibraryFactory> libraryFactories;
+
+	/**
+     * Will be populated by {@link }initLibraries()}
+     */
     private List<DefinitionLibrary<?>> libraries;
 
+	/**
+     * This should be called after the Spring context is initialized and libraryBeans and libraryFactories have been
+     * autowired (e.g. from this module's activator)
+     */
+    public void initLibraries() {
+        libraries = new ArrayList<DefinitionLibrary<?>>();
+        if (libraryBeans != null) {
+            libraries.addAll(libraryBeans);
+        }
+        if (libraryFactories != null) {
+            for (DefinitionLibraryFactory factory : libraryFactories) {
+                Collection<DefinitionLibrary<?>> fromFactory = factory.getDefinitionLibraries();
+                if (fromFactory != null) {
+                    libraries.addAll(fromFactory);
+                }
+            }
+        }
+    }
+
```

Comment:
Sure, I guess that's worth doing. (I was aiming to avoid any performance hit, but that can just be done via an if != null return